kimchi/src/prover_index.rs

```diff
@@ -91,6 +100,45 @@ where
             max_poly_size,
             max_quot_size,
             fq_sponge_params,
+            verifier_index: None,
+            verifier_index_digest: None,
+        }
+    }
+
+    /// Retrieve or compute the digest for the corresponding verifier index.
+    /// If the digest is not already cached inside the index, store it.
+    pub fn compute_verifier_index_digest<
+        EFqSponge: Clone + FqSponge<G::BaseField, G, G::ScalarField>,
+    >(
+        &mut self,
+    ) -> G::BaseField {
+        if let Some(verifier_index_digest) = self.verifier_index_digest {
+            return verifier_index_digest;
+        }
+
+        if let None = &self.verifier_index {
+            self.verifier_index = Some(self.verifier_index());
+        }
```

Comment:
why not just do `let verifier_index = self.verifier_index()`?

Reply:
Because then we don't get caching. This code is designed to only compute the `verifier_index` term when it's not already present.